lib/puppet/environments.rb

```diff
@@ -40,6 +40,10 @@ def get!(name)
     end
 
     def clear_all
+      # This method was implemented by future parser, but doesn't ever seem
+      # to be called, but might? be called via manual eviction.
+      root = Puppet.lookup(:root_environment) { nil }
+      root.instance_variable_set(:@static_catalogs, nil) unless root.nil?
```

Comment:
@hlindberg We did this for the future parser setting, but from what I can tell, the method is never called. Were you thinking we might add manual eviction in the (near) future, and so it's good to include this?